packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -68,6 +71,24 @@ export interface BarPlotProps extends Pick<BarElementProps, 'slots' | 'slotProps
    * Defines the border radius of the bar element.
    */
   borderRadius?: number;
+  /**
+   * Can be set to 'value' to display the current value.
+   * If provided, the function will be used to format the label of the bar.
+   * @param {BarItem} item The item to format.
+   * @param {BarLabelContext} context data about the bar.
+   * @returns {string} The formatted label.
+   */
+  barLabel?: 'value' | ((item: BarItem, context: BarLabelContext) => string | null | undefined);
```

Comment:
What's the purpose of `undefined` and `null`?

Reply:
Mostly to prevent the label from being displayed. If user returns `null|undefined` we are not displaying the label.

It also helps users by allowing them to return `item.value?.toString()` since bar values can be null.

Reply:
My bad, I miss read, I thought it was
```
'value' | ((item: BarItem, context: BarLabelContext) => string) | null | undefined;
```